From a050fd2fc95d7a11b34df407efe87ad96d6e0740 Mon Sep 17 00:00:00 2001
From: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
Date: Sun, 3 Jan 2021 04:21:30 +0100
Subject: [PATCH] PCI: dwc/intel-gw: enable the driver on Lantiq xRX200 and
 xRX300 SoCs

Lantiq xRX200 and xRX300 SoCs have an older variant of the Intel GW PCIe
IP core. Enable support for this older core with own compatible strings
and allow building the driver on that SoC architecture.

Signed-off-by: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
---
 drivers/pci/controller/dwc/Kconfig         | 12 ++++++------
 2 files changed, 8 insertions(+), 6 deletions(-)

--- a/drivers/pci/controller/dwc/Kconfig
+++ b/drivers/pci/controller/dwc/Kconfig
@@ -209,15 +209,16 @@ config PCIE_ARTPEC6_EP
 	  endpoint mode. This uses the DesignWare core.
 
 config PCIE_INTEL_GW
-	bool "Intel Gateway PCIe host controller support"
-	depends on OF && (X86 || COMPILE_TEST)
+	bool "Intel/Lantiq Gateway PCIe host controller support"
+	depends on OF && (X86 || SOC_TYPE_XWAY || COMPILE_TEST)
 	depends on PCI_MSI_IRQ_DOMAIN
 	select PCIE_DW_HOST
+	select REGMAP_MMIO
 	help
-	  Say 'Y' here to enable PCIe Host controller support on Intel
-	  Gateway SoCs.
-	  The PCIe controller uses the DesignWare core plus Intel-specific
-	  hardware wrappers.
+	  Say 'Y' here to enable PCIe Host controller support on Intel and
+	  Lantiq Gateway SoCs.
+	  The PCIe controller uses the DesignWare core plus Intel and Lantiq
+	  specific hardware wrappers.
 
 config PCIE_KIRIN
 	depends on OF && (ARM64 || COMPILE_TEST)
